pkg/helmclient/contract.go

```diff
@@ -0,0 +1,82 @@
+package helmclient
+
+import (
+	"time"
+
+	"helm.sh/helm/v3/pkg/release"
+)
+
+// ListOperation defines the contract for a list action
+type ListOperation struct {
+	AllNamespaces bool `json:"all-namespaces,omitempty"`
+	Deployed      bool `json:"deployed,omitempty"`
+	Failed        bool `json:"failed,omitempty"`
+	Pending       bool `json:"pending,omitempty"`
+	Uninstalled   bool `json:"uninstalled,omitempty"`
+	Uninstalling  bool `json:"uninstalling,omitempty"`
+
+	*GlobalFlags
```

Comment:
we shouldn't have pointer inside pointer struct. reference to flag struct doesn't need to be pointer in all operations.